src/main/java/io/jaegertracing/tests/CreateSpansRunnable.java

```diff
@@ -0,0 +1,151 @@
+
+import io.jaegertracing.internal.JaegerTracer;
+import io.jaegertracing.tests.model.TestConfig;
+import io.opentracing.Span;
+import io.opentracing.tag.Tags;
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author Pavol Loffay
+ */
+@Slf4j
+public class CreateSpansRunnable implements Runnable {
+    class SpansTimer extends TimerTask {
+
+        private long delay = 0;
+
+        public SpansTimer() {
+            if (config.isPerformanceTestLongRunEnabled()) {
+                // actual is 1000 ms, modify maximum available duration based on spans count.
+                // change it to microsecond
+                int maxDuration = 700;
+                if (config.getSpansCount() < 200) {
+                    maxDuration = 700;
+                } else if (config.getSpansCount() < 500) {
+                    maxDuration = 500;
+                } else {
+                    maxDuration = 100;
+                }
+
+                delay = (maxDuration * 1000L) / config.getSpansCount();
+                logger.debug("Maximun execution time per round:{}ms, delay between spans:{}us, spans count:{}",
+                        maxDuration, delay, config.getSpansCount());
+            } else {
+                delay = config.getPerformanceTestSpanDelay() * 1000L; // delay in microseconds
+            }
+        }
+
+        @Override
+        public void run() {
+            if (config.isPerformanceTestLongRunEnabled()
+                    && executedCount.get() >= config.getPerformanceTestDuration()) {
+                return;
+            }
+            Map<String, Object> logs = new HashMap<>();
+            logs.put("event", Tags.ERROR);
+            logs.put("error.object", new RuntimeException());
+            logs.put("class", this.getClass().getName());
+            int count = 0;
+            long startTime = System.currentTimeMillis();
+            do {
+                count++;
+                // emulate client spans
+                Span span = tracer.buildSpan(name)
+                        .withTag(Tags.COMPONENT.getKey(), "perf-test")
+                        .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT)
+                        .withTag(Tags.HTTP_METHOD.getKey(), "get")
+                        .withTag(Tags.HTTP_STATUS.getKey(), 200)
+                        .withTag(Tags.HTTP_URL.getKey(), "http://www.example.com/foo/bar?q=bar")
+                        .start();
+                span.log(logs);
+                span.finish();
+                try {
+                    TimeUnit.MICROSECONDS.sleep(delay);
+                } catch (InterruptedException ex) {
+                    logger.error("exception, ", ex);
+                }
+            } while (count < config.getSpansCount());
+            if (config.isPerformanceTestLongRunEnabled()) {
+                executedCount.incrementAndGet();
+                if (executedCount.get() % 60 == 0 || executedCount.get() == 1) { // print every 1 minute once
```

Comment:
I would rather remove this logs, a lot of unnecessary code with no value.

Reply:
I will increase the printing possibilities bit higher. I use this log to see all the spans are reported within a seconds. This may be useful when we running tests with `Long` running mode.